### PR TITLE
Package Elodie up as a Python PIP module

### PR DESCRIPTION
**Pull request by lukeplausin**
Hi Jaisen,

I read your blog about the Elodie project, and I really like the idea. I've been experimenting a little bit with Elodie, I think it's a great project and I'd like to contribute.

I made some changes on my fork to package Elodie up as a PIP module. This basically allows you to do some nice things using Pip:
* You can install Elodie and all python dependencies using a one line command, and without needing to use git (this is probably a bit nicer for newbies):
`pip install git+https://github.com/jmathai/elodie.git`
* The PBR packaging for pip includes CLI hooks, so you can invoke elodie as a CLI tool from any location:
```
cd (my pics folder)
elodie --source (my sd card) --destination .
```
* You can install elodie in developer mode to the current environment - this allows you to make changes on the fly without reinstalling the module
```
cd (elodie github repo)
pip install -e .
```
* There is the future possibility to upload Elodie to the Python module index, so that it can be downloaded with pip in the same was as other public modules

Anyway, please let me know if you like it. If not I can change a few things around. I tested the electron app also and it seems to be working.

Luke

### CODE REVIEW

File: elodie/tests/elodie_test.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 sys.path.insert(0, os.path.abspath(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.realpath(__file__)))))))
 
 import helper
-elodie = load_source('elodie', os.path.abspath('{}/../../elodie.py'.format(os.path.dirname(os.path.realpath(__file__)))))
+elodie = load_source('elodie', os.path.abspath('{}/../elodie.py'.format(os.path.dirname(os.path.realpath(__file__)))))
```
Comment (jmathai) [inline elodie/tests/elodie_test.py]: Is changing this path what's causing the tests to fail in Python 2.7?
https://travis-ci.org/jmathai/elodie/jobs/491249606

File: Readme.md [hunk 2]
```diff
@@ -29,12 +29,10 @@ dnf install perl-Image-ExifTool
 
 ### Clone the Elodie repository
 
-You can clone Elodie from GitHub. You'll need `git` installed ([instructions](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)).
+You can clone Elodie from GitHub. You'll need `pip` and `git` installed ([instructions](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)).
```
Comment (jmathai) [inline Readme.md]: Can update this to either clone from Github or install via pip. Don't need both.

File: .gitignore [hunk 3]
```diff
@@ -10,3 +10,129 @@ build/**
 **/*.nef
 **/*.rw2
 env/**
+
```
Comment (jmathai) [inline .gitignore]: Are most of these standard excludes or related to packaging for pip? Was just curious - no change needed.

Comment (lukeplausin) [reply .gitignore]: These are standard includes for python. The python compiler will add files to the repo which will get checked into git unless they're excluded by .gitignore

File: Readme.md (hunk 2 above)
Comment (lukeplausin) [reply Readme.md]: I believe you do still need git installed if you're installing via pip+git. If Elodie is accepted into the pypa public python repository then you wouldn't need git anymore.

File: elodie/tests/elodie_test.py (hunk 1 above)
Comment (lukeplausin) [reply elodie/tests/elodie_test.py]: I'm not exactly sure why the build is failing in 2.7, it doesn't seem to be an import error though. I haven't had time to investigate the 2.7 failure yet

**Comment by lukeplausin** [comment]
Hi @jmathai,

I think I may have fallen for the double import trap. When importing submodules from within the same module you are supposed to use dot notation instead of writing the name of the module. It wasn't a problem in Python 2.7 until I moved elodie.py into the elodie module folder, and it was never a problem in Python 3 since the module loading system was changed significantly as of Python 3.3.

http://python-notes.curiousefficiency.org/en/latest/python_concepts/import_traps.html

I managed to run the nose tests locally on 2.7 and 3.7 so I'm confident the build should pass this time.

LP

**Comment by lukeplausin** [comment]
Seems like the CLA checker is failing because I didn't have my git client configured for the correct email address. I tried to update the email address on my commits but it didn't seem to work. Not sure what I can do about that..

**Comment by jmathai** [comment]
@lukeplausin what a pain. Sorry about that. Have you tried this approach? https://stackoverflow.com/a/4982271

It is lesson learned from a prior project where I had to solicit CLAs from 100 contributors who had not signed one.

**Pull request closed without merging**